### DISCUSSION TITLE
DateRangePicker onDropdownClose value update

### DISCUSSION THREAD

**Discussion by abadire** [mantinedev/mantine#1833] [Q&A]
Hey! I need to set the end date of range as the next date from start if no end date is selected. Here is my solution:
```
<DateRangePicker
  {...form.getInputProps('dates')}
  onDropdownClose={() => {
    const [start, end] = form.values.dates;
    if (start && !end) {
      form.setFieldValue('dates', [start, addDays(start, 1)]);
    }
  }}
/>
```
which doesn't work, unfortunately. `form.values.dates` is not updated during `onDropdownClose`.

**Comment by rtivital** [comment]
You can update value in `useEffect`
